src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -363,32 +363,38 @@ public static bool IsForceRetainedAssembly (string assembly)
 
 		public static void SetWriteable (string source)
 		{
-			if (!File.Exists (source))
```

Comment:
Replacing a `File.Exists()` with a try/catch block just seems "icky".

And if callers should *already* be checking `File.Exists()`... is this in the "contract"?  "Documented" anywhere?  *Verified*?  And if all that is true, why have the try/catch here at all?

Reply:
I'm fine to let it throw 👍. I checked all the callers with `Find All References` in the IDE.

I should add a `<summary/>` mentioning this.

Reply:
It looks like `<ConvertResourcesCases/>` doesn't:

  * https://github.com/xamarin/xamarin-android/blob/720aa3c8e4e466339a0f08bdfba94cfbee5d88fb/src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs#L102

`Files.CopyIfChanged()` confuses me:

  * https://github.com/xamarin/xamarin-android/blob/720aa3c8e4e466339a0f08bdfba94cfbee5d88fb/src/Xamarin.Android.Build.Tasks/Utilities/Files.cs#L83-L101

  * `HasFileChanged()` can return true if *neither* `source` nor `destination` exists, meaning `File.Exists()` on either could be false.

Ditto for `Files.CopyIfBytesChanged()`, `Files.CopyIfStreamChanged()`, etc.

Reply:
This brings up a philosophical question.

Let's say we have:
```csharp
void Foo (string file)
{
    if (File.Exists (file)) {
        Bar (file);
        // Do something
    }
}

void Bar (string file)
{
    if (File.Exists (file)) {
        // Do something
    }
}
```

If `Foo` and `Bar` are both called a lot, let's say 1000's of times. What is the clean way to avoid the duplicate `File.Exists` call?

Do I need a `bool checkExists` parameter for `Bar`?